integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/PersonFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,455 @@
+		Person person = readResponse(response);
+		
+		// update the existing record
+		person.setId(WRONG_PERSON_UUID);
+		
+		// send the update to the server
+		response = put("/Person/" + WRONG_PERSON_UUID).jsonContent(toJson(person)).go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldUpdateExistingPersonAsXML() throws Exception {
+		// get the existing record
+		MockHttpServletResponse response = get("/Person/" + PERSON_UUID).accept(FhirMediaTypes.XML).go();
+		Person person = readResponse(response);
+		
+		// update the existing record
+		Date birthDate = DateUtils.truncate(new Date(), Calendar.DATE);
+		person.setBirthDate(birthDate);
+		
+		// send the update to the server
+		response = put("/Person/" + PERSON_UUID).xmlContext(toXML(person)).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		// read the updated record
+		Person updatedPerson = readResponse(response);
+		
+		assertThat(updatedPerson, notNullValue());
+		assertThat(updatedPerson.getIdElement().getIdPart(), equalTo(PERSON_UUID));
+		assertThat(updatedPerson.getBirthDate(), equalTo(birthDate));
+		assertThat(person, validResource());
+		
+		// double-check the record returned via get
+		response = get("/Person/" + PERSON_UUID).accept(FhirMediaTypes.XML).go();
+		Person reReadPerson = readResponse(response);
+		
+		assertThat(reReadPerson.getBirthDate(), equalTo(birthDate));
+	}
+	
+	@Test
+	public void shouldReturnBadRequestWhenDocumentIdDoesNotMatchPersonIdAsXML() throws Exception {
+		// get the existing record
+		MockHttpServletResponse response = get("/Person/" + PERSON_UUID).accept(FhirMediaTypes.XML).go();
+		Person person = readResponse(response);
+		
+		// update the existing record
+		person.setId(WRONG_PERSON_UUID);
+		
+		// send the update to the server
+		response = put("/Person/" + PERSON_UUID).xmlContext(toXML(person)).go();
+		
+		assertThat(response, isBadRequest());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldDeleteExistingPerson() throws Exception {
+		MockHttpServletResponse response = delete("/Person/" + PERSON_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
```

Comment:
@varung-31 when i look in to the current error ie 
```

[ERROR]   PersonFhirResourceProviderIntegrationTest.shouldDeleteExistingPerson:344                                                                                                               Expected: response with HTTP status indicating request was handled successfully                                                                                                                       but: response with status code <410> with message "61b38324-e2fd-4feb-95b7-9e9a2a4400df"
--
```
throw on this method the uuid is deleted successfully, thats why we have  `code <410> with message "61b38324-e2fd-4feb-95b7-9e9a2a4400df"` tho am not quite sure as to why the response message here is negative 
--

Reply:
Admittedly, we should improve the message here somewhat, especially as I don't think that response should be returned from a `delete()` operation. So what happens in [`delete()`](https://github.com/openmrs/openmrs-module-fhir2/blob/6724bedc2730c379e6904fc7544916834014689e/api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseFhirDao.java#L125) is first we `get()` the result. In [`get()`](https://github.com/openmrs/openmrs-module-fhir2/blob/6724bedc2730c379e6904fc7544916834014689e/api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseFhirDao.java#L96), if the result is retired or voided already, we throw an exception. This is to ensure that an attempt to `get()` a record after it's been delete generates an error correctly.

The short version is that you're essentially asking to delete an already-voided Person record, so maybe try testing the functionality with a different Person. Nonetheless, that's a valuable error to have brought out.

Reply:
@gitcliff i don't think that is the case. A 410 error means that the resource you are trying to delete is GONE. I think it might have been voided somewhere. We need to check where the bug is.